src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -858,7 +858,7 @@ internal static unsafe int Read(SafeFileHandle fd, byte[] buffer, int offset, in
             fixed (byte* bufPtr = buffer)
             {
                 int result;
-                while (Interop.CheckIo(result = Interop.Sys.Read(fd, (byte*)bufPtr + offset, count))) ;
+                Interop.CheckIo(result = Interop.Sys.Read(fd, (byte*)bufPtr + offset, count));
```

Comment:
Similar to a previous comment, if CheckIo returned its first input, this could be:
```C#
int result = Interop.CheckIo(Interop.Sys.Read(fd, (byte*)bufPtr + offset, count)));
```
Seems a bit cleaner.